### PR TITLE
cgo: Add rescan from height.

### PR DESCRIPTION
**Pull request by JoeGruffins**
Will probably add the rest of the methods in this pr. atm I think it's just this and sign message.

### CODE REVIEW

**Comment by JoeGruffins** [comment]
Something weird about newExternalAddress, if you get the currentReceiveAddress first, it does not update the next newExternalAddress. So I guess new external will give you the current address before updating the index. Maybe we need to track that internally so that we can always return the next address for new.

File: asset/dcr/loader.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ import (
 
 const (
 	walletDbName = "wallet.db"
+	entropyBytes = 18 // 144 bits
```
Comment (buck54321) [inline asset/dcr/loader.go:18]: Can export in dcrdex if you'd like.

File: cgo/addresses.go [hunk 2]
```diff
@@ -24,3 +28,100 @@ func currentReceiveAddress(cName *C.char) *C.char {
...
+		return errCResponseWithCode(ErrCodeNotSynced, "newExternalAddress requested on an unsynced wallet")
+	}
+
+	_, err := w.NewExternalAddress(ctx, udb.DefaultAccountNum)
+	if err != nil {
+		return errCResponse("w.NewExternalAddress error: %v", err)
+	}
+
+	// NewExternalAddress will take the current address before increasing
+	// the index. Get the current address after increasing the index.
```
Comment (buck54321) [inline cgo/addresses.go:50]: How weird.

**Review by buck54321 — APPROVED** [review]
LGTM

**Pull request closed without merging**